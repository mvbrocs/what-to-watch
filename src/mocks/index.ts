export { default as mockFilms } from './films';
export { default as mockStore } from './store';
