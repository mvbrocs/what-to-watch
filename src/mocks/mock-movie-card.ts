import { mockMovies } from './mock-movies';

export const mockMovieCard = mockMovies[0];
