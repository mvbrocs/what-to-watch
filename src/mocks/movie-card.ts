import { movies } from './movies';

export const movieCardMock = movies[0];
