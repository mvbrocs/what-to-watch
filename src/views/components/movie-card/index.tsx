export type MovieCardProps = {
  name: string;
  preview_image: string;
  preview_video_link: string;
};
export { MovieCard } from './movie-card';
