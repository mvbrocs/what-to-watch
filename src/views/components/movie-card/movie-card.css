.small-movie-card--default .small-movie-card__image {
  height: auto;
}

.small-movie-card--default .small-movie-card__image video {
  position: absolute;
  top: 50%;
  left: 50%;
  min-width: 100%;
  min-height: 100%;
  transform: translate(-50%, -50%);
}

.small-movie-card--default .small-movie-card__image img {
  position: relative;
}

.small-movie-card__image--img-hidden img {
  visibility: hidden;
}
