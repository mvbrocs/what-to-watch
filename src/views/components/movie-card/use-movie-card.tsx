import { useState } from 'react';

export const useMovieCard = () => {
  const [mouseEnter, setMouseEnter] = useState(false);
  const cardMouseEnterHandler = () => setMouseEnter(true);
  const cardMouseLeaveHandler = () => setMouseEnter(false);

  return { mouseEnter, cardMouseEnterHandler, cardMouseLeaveHandler };
};
