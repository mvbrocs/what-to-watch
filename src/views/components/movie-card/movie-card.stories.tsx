import React from 'react';

import { MovieCard } from './';
import { movies } from 'mocks/movies';

export default {
  component: MovieCard,
  title: 'MovieCard',
  excludeStories: /.*Data$/,
};

export const movieCardData = movies[0];

export const Default = () => <MovieCard {...movieCardData} />;
