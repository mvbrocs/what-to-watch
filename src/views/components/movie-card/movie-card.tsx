import React from 'react';

import { VideoPlayer } from 'views/components/video-player/video-player';
import { IMovie } from 'mocks/movies';
import { useMovieCard } from './use-movie-card';

type MovieCardProps = IMovie;

const VIDEO_MS_DELAY = 1000;

export const MovieCard = ({ title, img, videoSrc }: MovieCardProps) => {
  const { mouseEnter, cardMouseEnterHandler, cardMouseLeaveHandler } = useMovieCard();

  return (
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={cardMouseEnterHandler}
      onMouseLeave={cardMouseLeaveHandler}
    >
      <div className="small-movie-card__image">
        <VideoPlayer play={mouseEnter} src={videoSrc} poster={img} delay={VIDEO_MS_DELAY} />
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">
          {title}
        </a>
      </h3>
    </article>
  );
};
