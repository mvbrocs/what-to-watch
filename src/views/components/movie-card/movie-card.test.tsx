import React from 'react';
import { shallow } from 'enzyme';

import { MovieCard } from './';
import { movieCardMock } from 'mocks/movie-card';

describe('MovieCard', () => {
  it('renders properly', () => {
    const movieCard = shallow(<MovieCard {...movieCardMock} />);

    expect(movieCard).toMatchSnapshot();
  });
});
