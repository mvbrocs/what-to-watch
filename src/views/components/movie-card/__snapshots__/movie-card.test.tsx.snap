// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders properly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <VideoPlayer
      delay={1000}
      play={false}
      poster={
        Object {
          "alt": "Fantastic Beasts: The Crimes of Grindelwald",
          "height": 175,
          "src": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
          "width": 280,
        }
      }
      src="https://st.depositphotos.com/1100827/3666/v/600/depositphotos_36660843-stock-video-girl-leans-back-among-the.mp4"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
